docs/v1-proposed-api.md

```diff
@@ -0,0 +1,280 @@
+# Proposed Changes for Hive V1 API
+
+Minimal changes, no new CRDs or relationships between.
+
+## ClusterDeployment
+
+### Current V1Alpha1 ClusterDeployment Example
+
+```yaml
+apiVersion: hive.openshift.io/v1alpha1
+kind: ClusterDeployment
+metadata:
+  name: ci-cluster-v4-1
+  namespace: uhc-staging-18urm2sbab85d9crlk27is39n706ph3g
+spec:
+  baseDomain: r4f1.s1.devshift.org
+  certificateBundles:
+  - generate: true
+    name: primary-cert-bundle
+    secretRef:
+      name: primary-cert-bundle-secret
+  clusterName: ci-cluster-v4-1
+  compute:
+  - name: worker
+    platform:
+      aws:
+        rootVolume:
+          iops: 100
+          size: 300
+          type: gp2
+        type: m5.xlarge
+        zones:
+        - us-east-1a
+    replicas: 4
+  controlPlane:
+    name: master
+    platform:
+      aws:
+        rootVolume:
+          iops: 100
+          size: 32
+          type: gp2
+        type: m5.xlarge
+        zones:
+        - us-east-1a
+    replicas: 3
+  controlPlaneConfig:
+    servingCertificates:
+      default: primary-cert-bundle
+  imageSet:
+    name: openshift-v4.1.18
+  images: {}
+  ingress:
+  - domain: apps.ci-cluster-v4-1.r4f1.s1.devshift.org
+    name: default
+    servingCertificate: primary-cert-bundle
+  installed: true
+  manageDNS: true
+  networking:
+    clusterNetworks:
+    - cidr: 10.128.0.0/14
+      hostSubnetLength: 23
+    machineCIDR: 10.0.0.0/16
+    serviceCIDR: 172.30.0.0/16
+    type: OpenShiftSDN
+  platform:
+    aws:
+      region: us-east-1
+  platformSecrets:
+    aws:
+      credentials:
+        name: aws
+  pullSecret:
+    name: pull
+  sshKey:
+    name: ssh
+status:
+  adminKubeconfigSecret:
+    name: ci-cluster-v4-1-0-ndv7l-admin-kubeconfig
+  adminPasswordSecret:
+    name: ci-cluster-v4-1-0-ndv7l-admin-password
+  apiURL: https://api.ci-cluster-v4-1.r4f1.s1.devshift.org:6443
+  certificateBundles:
+  - generated: true
+    name: primary-cert-bundle
+  cliImage: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:528f2ead3d1605bdf818579976d97df5dd86df0a2a5d80df9aa8209c82333a86
+  clusterID: 1e4f00a3-a30e-46d7-9567-4bb61caf6d73
+  clusterVersionStatus:
+    [SNIP]
+  conditions:
+  - lastProbeTime: "2019-10-23T08:40:24Z"
+    lastTransitionTime: "2019-10-23T08:40:24Z"
+    message: Control plane certificates are present
+    reason: ControlPlaneCertificatesFound
+  installed: true
+  installedTimestamp: "2019-10-23T08:38:03Z"
+  installerImage: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:c9c1e6d1897b74d366d331f5cdc3fa16c5a96abe28ff8238833c09042156e983
```

Comment:
Why are cliImage and installerImage used rather than just the release image pullspec, from which the installer, cli, and everything else can be gathered?

Reply:
These are actually gathered, they're in status here, and are set by a pod that runs and extracts them from the release image. We just store what was used for posterity mostly.